### ISSUE TITLE
Clima takes time to load.

### ISSUE THREAD

**Issue by Suvrat31** [londonappbrewery/Flutter-Course-Resources#19]
The final build output of the app doesnt load the weather from the OpenWeatherMap API and is stuck on the loading screen, while it works when the app is in testing mode. There are no network issues or GPS issues with my device. Kindly check the app by installing it from the zip file below.
[app-release.zip](https://github.com/londonappbrewery/Flutter-Course-Resources/files/4583163/app-release.zip)

**Comment by gupta-shrinath** [comment]
Can you provide your code, please?

**Comment by brothersully** [comment]
same issue man. I've been banging my head over this. Have you found a solution??

**Comment by Suvrat31** [comment]
> Can you provide your code, please?

It is the same code they use in the videos. You can check out their final solution codes repo

**Comment by Suvrat31** [comment]
> same issue man. I've been banging my head over this. Have you found a solution??

@brothersully Nah, nothing has worked out for me yet.

**Comment by gupta-shrinath** [comment]
You have created your own  OpenWeatherMap API, right? Can you provide the console output?

**Comment by brothersully** [comment]
> @brothersully Nah, nothing has worked out for me yet.

@Suvrat31 I've got some good news for you. I've finally solved this issue. On your network.dart file you must've used a line `Geolocator()..forceAndroidLocationManager = true;`
The emulator doesn't return the location without this line of code but the app does. This solved the issue. Hope it helps you too! If it doesn't let me know.

**Comment by moumitb** [comment]
Anyone got any solution, for this issue?

**Comment by gupta-shrinath** [comment]
@moumitb You have created your own OpenWeatherMap API, right? Can you provide the console output?

**Comment by Kovalchik8** [comment]
Emulator:
1. Open settings -> Location -> Set location
2. Open google maps and locate your device
3. Reload the app

Physical Android devise: 
1. Goto android/app/src/main/AndroidManifest.xml and add <uses-permission android:name="android.permission.INTERNET"/> just before the <application> (https://stackoverflow.com/questions/62261690/flutter-app-gets-stuck-at-loading-screen)

**Comment by Aatashh** [comment]
F/crash_dump32(23307): crash_dump.cpp:246] target died before we could attach (received main tid = 23305)

what is this error on my consol?

**Comment by gupta-shrinath** [comment]
Maybe Aatashh your device got disconnected

**Comment by Aatashh** [comment]
> Maybe Aatashh your device got disconnected

I am using Emulator :(

**Comment by gupta-shrinath** [comment]
Is it happening recursively?

**Comment by Aatashh** [comment]
> Is it happening recursively?

yes even after fully closing and opening the android studio, I am not missing anything and I am comparing the code and its 100% matching

**Comment by gupta-shrinath** [comment]
Can you use a physical device?
Try to invalidate cache  and restart the android studio

**Comment by Aatashh** [comment]
> Can you use a physical device?
> Try to invalidate cache and restart the android studio

Just connected the phone physically and it worked :) thanks

**Comment by Bhavay192** [comment]
> The final build output of the app doesnt load the weather from the OpenWeatherMap API and is stuck on the loading screen, while it works when the app is in testing mode. There are no network issues or GPS issues with my device. Kindly check the app by installing it from the zip file below.
> [app-release.zip](https://github.com/londonappbrewery/Flutter-Course-Resources/files/4583163/app-release.zip)

Having the same problem i even tried the internet permission still the app gets stuck at the load screen . :(

**Comment by Bhavay192** [comment]
plss help me with a suitable solution

**Comment by MuhammadSabah** [comment]
Anyone found it? i have the same problem aswell

**Comment by Kovalchik8** [comment]
> Anyone found it? i have the same problem aswell

Have you read my answer above? I wonder why it hasn't helped anybody yet.

**Comment by MuhammadSabah** [comment]
the Set Location button is disabled i cant click on it

**Comment by MuhammadSabah** [comment]
A fixed device location but still it doesn't go to the next page 😔

**Comment by aditmo** [comment]
You need to provide internet access to your app by enabling it in AndroidManifest.xml . So go to app -> src -> main -> AndroidManifest.xml. and paste this code-  `<uses-permission android:name="android.permission.INTERNET" />`

**Comment by aditmo** [comment]
> The final build output of the app doesnt load the weather from the OpenWeatherMap API and is stuck on the loading screen, while it works when the app is in testing mode. There are no network issues or GPS issues with my device. Kindly check the app by installing it from the zip file below.
> [app-release.zip](https://github.com/londonappbrewery/Flutter-Course-Resources/files/4583163/app-release.zip)

You need to provide internet access to your app by enabling it in AndroidManifest.xml . So go to app -> src -> main -> AndroidManifest.xml. and paste this code- <uses-permission android:name="android.permission.INTERNET" />

**Comment by aditmo** [comment]
> The final build output of the app doesnt load the weather from the OpenWeatherMap API and is stuck on the loading screen, while it works when the app is in testing mode. There are no network issues or GPS issues with my device. Kindly check the app by installing it from the zip file below.
> [app-release.zip](https://github.com/londonappbrewery/Flutter-Course-Resources/files/4583163/app-release.zip)

You need to provide internet access to your app by enabling it in AndroidManifest.xml . So go to app -> src -> main -> AndroidManifest.xml. and paste this code- `uses-permission android:name="android.permission.INTERNET" />`

**Comment by raunakmitra1** [comment]
Hi guys,

I was facing this problem and I added <uses-permission android:name="android.permission.INTERNET" /> to my AndroidManifest.xml file. Also in the location.dart file, in the services folder I changed the Geolocator positioning from low to high. I tried setting it to medium, but that did not work and the app remained stuck in the loading screen. Only when I changed it to high did I get the app start working correctly.

**Initial**
      Position position = await Geolocator().getCurrentPosition(
          desiredAccuracy: LocationAccuracy.low);

**Changed**
      Position position = await Geolocator().getCurrentPosition(
          desiredAccuracy: LocationAccuracy.high);

**Comment by Angelo-Al** [comment]
Hi!

I was also facing this problem, but no solution worked. As the issue is still open, if none of the solutions from above worked, then here's what I've found. _(note: this works if you haven't changed the \_appid under weather.dart from the source code)_

Check your console and see if there's any unhandled exception. I didn't notice it at first, but in my case, the openwheathermap api gave an error:  `[ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: HttpException: Responded with 401, uri = https://api.openweathermap.org/data/2.5/weather?appid=c029cfbbee8e7d8e8aa732f185cddd4a&units=metric&lat=14.6912167&lon=121.02661`

The API key of the original code probably expired already (or something similar to that). So what you need to do is the following:
1. Go to [https://openweathermap.org/price](https://openweathermap.org/price) and set up a Free subscription account.
2. After the setup, on the NavBar > Account tab > My API Keys ![image](https://user-images.githubusercontent.com/63029417/141291357-f741e3ae-34cf-4bf3-b974-db02a19fbbd5.png)
3. Get the Key (copy it)
4. Under weather.dart, replace the value of `static const _appid` with the Key (select the string and paste it)
5. Restart the application

Important things to note:
1. Doesn't accurately display the City name sometimes.
2. Solution by raunamiktra1 has been implemented. 
      > Initial
      > Position position = await Geolocator().getCurrentPosition(
      > desiredAccuracy: LocationAccuracy.low);
      > 
      > Changed
      > Position position = await Geolocator().getCurrentPosition(
      > desiredAccuracy: LocationAccuracy.high);
3. Solution by Kovalchik8 has been implemented.
      > Goto android/app/src/main/AndroidManifest.xml and add `<uses-permission android:name="android.permission.INTERNET"/>` just before the `<application>` tag (https://stackoverflow.com/questions/62261690/flutter-app-gets-stuck-at-loading-screen)




[Here's where I got the idea of why mine didn't work](https://github.com/kalliope-project/kalliope_neuron_openweathermap/issues/3)
